### PR TITLE
IllusionLib

### PR DESCRIPTION
**Pull request by mtesija**
Right now Arachna is the only bot that is setup to use illusionLib.

Do we want all of the bots to have access to this? or just specified bots like Arachna?

In addition, I used an override to onthink to implement this because it needs access to core, botbrain, and behavioLib. If you know of a better way to do this let me know and I'll change it.

### CODE REVIEW

**Comment by Kairus101** [comment]
I'd recommend everyone uses it, ie add it to botBrainCore
Bots still pick up illusion runes etc

**Comment by Kairus101** [comment]
I am unable to come up with profiling numbers to see how much of an impact this code has performance-wise, so I will have to eyeball it for the meantime.. :) but it doesn't seem too costly.

File: bots/illusionLib.lua [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+                end
+        end
+
+        return
+end
+
+function object:illusionLibOnthinkOverride(tGameVariables)
+        self:illusionLibOnthinkOld(tGameVariables)
+
+        if illusionLib.nNextBehaviorTime <= HoN.GetGameTime() then
```
Comment (Kairus101) [inline bots/illusionLib.lua]: if no illusions are found, this function will run every frame
illusionLib.nNextBehaviorTime seems to be set in the wrong place

**Comment by mtesija** [comment]
Alright it is completely set up now, and should work for every hero. Although I've only tested it on a small number of them.

I also updated the illusion checking function to use tControllableUnits based on Anakonda's suggestion.

Finally I added illusionLib.bRunBehaviors so that people can completely disable the behaviors portion of illusionLib

**Comment by Kairus101** [comment]
Not able to test until tomorrow around this time, where I'll likely merge it :)

**Comment by Kairus101** [comment]
argh, work popped up, won't be free for another couple of days

**Pull request merged**